### PR TITLE
Release task after setting err in panic/recover block to avoid race on err variable (AO-18198)

### PR DESCRIPTION
**Pull request by lootek**
JIRA: AO-18198

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,6 +7,8 @@ require (
 	github.com/julienschmidt/httprouter v1.3.0
 	github.com/sirupsen/logrus v1.7.0
 	github.com/smartystreets/goconvey v1.6.4
+	github.com/solarwinds/snap-plugin-lib/v2 v2.0.2 // indirect
```
Comment (adamiklukasz) [inline go.mod:10]: this is strange? reference to the itself?

File: go.sum [hunk 2]
```diff
@@ -5,6 +5,7 @@ github.com/client9/misspell v0.3.4/go.mod h1:qj6jICC3Q7zFZvVWo7KLAzC3yx5G7kyvSDk
 github.com/cncf/udpa/go v0.0.0-20191209042840-269d4d468f6f/go.mod h1:M8M6+tZqaGXZJjfX53e64911xZQV5JYwmTeXPW+k8Sc=
 github.com/cpuguy83/go-md2man/v2 v2.0.0-20190314233015-f79a8a8ca69d h1:U+s90UTSYgptZMwQh2aRr3LuazLJIa+Pg3Kc1ylSYVY=
 github.com/cpuguy83/go-md2man/v2 v2.0.0-20190314233015-f79a8a8ca69d/go.mod h1:maD7wRr/U5Z6m/iR4s+kqSMx2CaBsrgA7czyZG/E6dU=
+github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
```
Comment (tomekkolo) [inline go.sum:8]: go mod tidy

**Review by adamiklukasz — DISMISSED** [review]

**Review by tomekkolo — APPROVED** [review]

File: go.sum (hunk 2 above)
Comment (lootek) [reply go.sum:10]: done

File: go.mod (hunk 1 above)
Comment (lootek) [reply go.mod]: good catch... I've run go mod update and tidy and it's gone now

**Review by adamiklukasz — APPROVED** [review]

**Review by tomekkolo — APPROVED** [review]

**Pull request merged**